frontend/src/app/analytics/Analytics.test.tsx

```diff
@@ -279,15 +281,18 @@ describe("Analytics", () => {
   });
   it("allows filtering by a custom date range", async () => {
     await screen.findByText("COVID-19 testing data");
-    userEvent.selectOptions(screen.getByLabelText("Date range"), [
+    await userEvent.selectOptions(screen.getByLabelText("Date range"), [
       "Custom date range",
     ]);
     await screen.findByText("COVID-19 testing data");
     const startDate = screen.getByTestId("startDate") as HTMLInputElement;
     const endDate = screen.getByTestId("endDate") as HTMLInputElement;
-    userEvent.type(startDate, "2021-07-01");
+
+    fireEvent.change(startDate, { target: { value: "2021-07-01" } });
```

Comment:
question: is there a reason we want the fireEvent API here rather than the higher-level userEvent? Guessing that the test was flaking for some reason with `userEvent.type`, but wanted to confirm / add context here since [the docs call out this difference explicitly](https://testing-library.com/docs/dom-testing-library/api-events/)

Reply:
user event library is not working correctly for inputs of type date. Sometimes the inputs do not get updated.  Changing the type event back to the fireEvent library makes the tests less flaky

Reply:
I did come across [this comment](https://github.com/testing-library/user-event/issues/399#issuecomment-815664027) in the RTL repo about `userEvent` needing to be `.clear()`-ed first for date inputs. Tested it locally and it did seem to fix flakeage.

No need to fix it for this PR, but thought I'd leave it here for the record.